python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -48,6 +49,78 @@ def destroy_engine(self):
         if self.msg_server is not None:
             self.msg_server.close()
 
+
+    ###########################################################################
+    # panel support (displayed as tabs)
+    def _init_tabs(self):
+        """
+        Create desktop tabs from the registered application panels, following
+        their specified relative positioning.
+        """
+        # make the "Apps" tab appear as a fake panel to fit in the sorting
+        # logic
+        apps_tab = {"callback":   self.desktop_window._register_apps_tab,
+                    "properties": {"position": 0}}
+
+        panels = [apps_tab] + list(self._engine.panels.itervalues())
+
+        def panel_position(panel):
+            """ Returns the relative position of a panel """
+            # sane checking on the expected structure
+            assert("properties" in panel, "panel should have properties set")
+            assert("position" in panel["properties"],
+                   "panel should have a position set")
+            return panel["properties"]["position"]
+
+        ordered_panels = sorted(panels, key=panel_position)
```

Comment:
I think it is ok.  The apps themselves come in as a dictionary from yaml, which has the same ordering issue right from the get go.  Having position at least gives some control over and we can add a feature where we have a way via settings on desktop to create an explicit ordering, which will probably be what we do if clients do start making their own apps or we end up with a bunch of panel apps.  Probably overkill right now.